File: mobile-app/app/src/main/java/com/fenix/streaming/config/ApiConfig.kt
package com.fenix.streaming.config

/**
 * API Configuration for Fenix Streaming Backend
 * 
 * Update FENIX_API_BASE_URL with your backend URL:
 * - Development: http://localhost:5000 (emulator: http://10.0.2.2:5000)
 * - Production: https://your-fenix-backend.com
 */
object ApiConfig {
    // ⚠️ UPDATE THIS WITH YOUR FENIX BACKEND URL
    const val FENIX_API_BASE_URL = "http://10.0.2.2:5000"
    
    // API Endpoints
    const val LOGIN_ENDPOINT = "/api/auth/login"
    const val MOVIES_ENDPOINT = "/api/movies"
    const val USERS_ENDPOINT = "/api/app-users"
    
    // JWT Token timeout (7 days in milliseconds)
    const val TOKEN_EXPIRY_MS = 7L * 24 * 60 * 60 * 1000
    
    // Retry configuration
    const val CONNECT_TIMEOUT_SECONDS = 30
    const val READ_TIMEOUT_SECONDS = 30
    const val WRITE_TIMEOUT_SECONDS = 30
}

File: fenix-flutter/lib/config/api_config.dart
/// API Configuration for Fenix Backend
/// 
/// Update [apiBaseUrl] with your Fenix backend URL:
/// - Development: http://localhost:5000
/// - Production: https://your-fenix-backend.com

class ApiConfig {
  //  UPDATE THIS WITH YOUR FENIX BACKEND URL
  static const String apiBaseUrl = 'https://e1c30eb1-234f-4841-89d1-b4cd3d839858-00-168qrarj2xv9x.spock.replit.dev';
  
  // API Endpoints
  static const String loginEndpoint = '/api/auth/login';
  static const String moviesEndpoint = '/api/movies';
  static const String usersEndpoint = '/api/app-users';
  
  // JWT Token timeout (7 days in milliseconds)
  static const int tokenExpiryMs = 7 * 24 * 60 * 60 * 1000;
  
  // HTTP Timeouts
  static const int connectTimeoutSeconds = 30;
  static const int readTimeoutSeconds = 30;
}
